### PR TITLE
Add SimpleXmlElementXpathMethodReturnTypeExtension

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

File: src/Type/Php/SimpleXMLElementXpathMethodReturnTypeExtension.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		}
+
+		$argType = $scope->getType($methodCall->args[0]->value);
+
+		$xmlElement = new \SimpleXMLElement('<foo />');
+
+		$result = null;
+		foreach (TypeUtils::getConstantStrings($argType) as $constantString) {
+			set_error_handler(null);
+			$newResult = $xmlElement->xpath($constantString->getValue());
```
Comment (ondrejmirtes) [inline src/Type/Php/SimpleXMLElementXpathMethodReturnTypeExtension.php:44]: What about `@` here?

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Type/Php/SimpleXMLElementXpathMethodReturnTypeExtension.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Type/Php/SimpleXMLElementXpathMethodReturnTypeExtension.php]: It seems to work, but some randoms tests are failing...

File: src/Type/Php/SimpleXMLElementXpathMethodReturnTypeExtension.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+		if ($result === null || !$argType instanceof NeverType) {
+			return ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
+		}
+
+		if ($result === false) {
+			return new ConstantBooleanType(false);
+		}
+
+		return new ArrayType(new MixedType(), new StaticType(\SimpleXMLElement::class));
```
Comment (ondrejmirtes) [inline src/Type/Php/SimpleXMLElementXpathMethodReturnTypeExtension.php:61]: The 2nd argument shouldn't be new StaticType, but `$scope->getType($methodCall->var)` instead.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Type/Php/SimpleXMLElementXpathMethodReturnTypeExtension.php (hunk 2 above)
Comment (VincentLanglet) [reply src/Type/Php/SimpleXMLElementXpathMethodReturnTypeExtension.php]: Done ! :)

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**